### PR TITLE
Dismiss with core cancel

### PR DESCRIPTION
**Pull request by olmokramer**
Listen to core:cancel on atom-workspace. When triggered dismiss notifications.

Moved here from atom/atom#5972

### CODE REVIEW

File: lib/main.coffee [hunk 1]
```diff
@@ -39,6 +39,10 @@ Notifications =
           stack: originalError.stack
           dismissable: true
         atom.notifications.addFatalError(message, options)
+        
+    @subscriptions.add atom.commands.add 'core:cancel': ->
+      atom.notifications.getNotifications().forEach (notification) ->
```
Comment (benogle) [inline lib/main.coffee]: Can you just use a loop? No reason to use forEach here.

Comment (olmokramer) [inline lib/main.coffee]: Sure

**Comment by benogle** [comment]
Also, please add a spec

**Comment by olmokramer** [comment]
Spec added :)

**Comment by benogle** [comment]
Thanks for this.

**Comment by bwagner** [comment]
I'm on Atom 1.41.0, using [vim-mode-plus](https://atom.io/packages/vim-mode-plus) 1.36.3.
In this case, pressing ESC does not dismiss notification, but the ESC key is consumed by the vim mode.

**Pull request merged**